lib/modules/manager/github-actions/extract.ts

```diff
@@ -86,6 +88,45 @@ function extractContainer(container: unknown): PackageDependency | undefined {
   return undefined;
 }
 
+const runnerVersionRegex = regEx(
+  /^\s*(?<depName>[a-zA-Z]+)-(?<currentValue>[^\s]+)/
+);
+
+function extractRunner(runner: string): PackageDependency | undefined {
+  const runnerVersionGroups = runnerVersionRegex.exec(runner)?.groups;
+  if (!runnerVersionGroups) {
+    return;
+  }
+
+  const { depName, currentValue } = runnerVersionGroups;
+
+  const dependency: PackageDependency = {
```

Comment:
we should return null if depName isn't a known runner, otherwise it extracts `abc-123` ad valid dependency and later it fails with lookup warning.

you can use the datasource static values to check if it's valid, so we don't need to maintain multiple places.